fixed the external data checker from stalling when git tag pattern is incorrect, this should fix the github actions run

tobiasmicheler opened the pull request:
I recently opened a related issue:
This closes https://github.com/flathub-infra/flatpak-external-data-checker/issues/418
In this issue I said, that I don't know how to fix this problem. However, I figured it out.

wjt commented:
Thanks, this does fix the problem. Nice work tracking this down!

I'll amend your commit slightly for style.

What I am trying to work out is: why does the assertion failure hang the asyncio loop forever?

wjt committed:
htmlchecker: Move _get_pattern() helper to Checker class

This will be used to replace some very similar code in gitchecker.

wjt committed:
gitchecker: Use Checker._get_pattern helper

Rather than asserting when given the wrong number of match
groups, this helper raises a descriptive exception. This is both more
helpful to the developer who has incorrectly specified the tag-pattern,
but also seems to prevent an issue where an assertion failure causes the
asyncio mainloop to hang.

Fixes #418

Thanks to Tobias Micheler <tobiasmicheler@outlook.de> for finding the
issue!
